test-suite/basketoption.cpp

```diff
@@ -1105,13 +1106,15 @@ test_suite* BasketOptionTest::suite(SpeedLevel speed) {
     suite->add(QUANTLIB_TEST_CASE(&BasketOptionTest::test2DPDEGreeks));
 
     if (speed <= Fast) {
-        const Size nTestCases = std::min(Size(5), LENGTH(oneDataValues));
-        for (Size i=0; i < nTestCases; ++i) {
-            suite->add(QUANTLIB_TEST_CASE(
-                boost::bind(&BasketOptionTest::testOneDAmericanValues,
-                    (i    *LENGTH(oneDataValues))/nTestCases,
-                    ((i+1)*LENGTH(oneDataValues))/nTestCases)));
-        }
+        #define N_TEST_CASES 5
```

Comment:
What was the problem here? Just curious.

Reply:
with the upcoming boost_1_67 release it is no longer possible that two test cases have the same name. 

If no name is provided boost more or less takes the method name as internal test case name and due to the usage of boost::bind here all test cases within the for loop got the same internal name. 

This changes with the usage of the preprocessor for loop. I was also considering to remove the test case split here and to get back to the old solution where we have had one test case running for multiple minutes.

Reply:
I see, thanks.

Reply:
(And it's ok to keep the test cases split.)